identity_presentation/src/commonMain/kotlin/net/thechance/mena/identity/presentation/screen/pickLocation/PickLocationScreenViewModel.kt

```diff
@@ -0,0 +1,196 @@
+package net.thechance.mena.identity.presentation.screen.pickLocation
+
+import androidx.compose.ui.unit.DpOffset
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.IO
+import net.thechance.mena.identity.domain.repository.MobileLocationRepository
+import net.thechance.mena.identity.domain.util.Coordinates
+import net.thechance.mena.identity.presentation.base.BaseScreenModel
+import net.thechance.mena.identity.presentation.base.ErrorState
+import net.thechance.mena.identity.presentation.mapper.mapErrorToMessage
+import net.thechance.mena.identity.presentation.util.permissionHandler.PermissionHandler
+import net.thechance.mena.identity.presentation.util.permissionHandler.PermissionState
+import org.maplibre.compose.camera.CameraPosition
+
+class PickLocationScreenViewModel(
+    private val mobileLocationRepository: MobileLocationRepository,
+    private val dispatcher: CoroutineDispatcher = Dispatchers.IO,
+    private val locationForegroundHandler: PermissionHandler,
+    private val addressModel: AddressModel?,
+) : BaseScreenModel<PickLocationScreenUIState, PickLocationScreenUIEffect>(PickLocationScreenUIState()),
+    PickLocationScreenInteractionListener {
+    init {
+        if (addressModel != null) {
+            updateState {
+                copy(
+                    currentLocation = Coordinates(addressModel.latitude, addressModel.longitude).toUiState(),
+                    address = addressModel.addressLine,
+                    isMapLocked = true,
+                    animateToCurrentLocation = true,
+                )
+            }
+            changeIsConfirmEnabled()
+        }
+    }
+
+        override fun onClickMap(
+            coordinates: PickLocationScreenUIState.CoordinatesUiState,
+            pointerLocation: DpOffset
+        ) {
+            updateState {
+                copy(
+                    currentLocation = coordinates,
+                    pointerLocation = pointerLocation,
+                    isMapLocked = true
+                )
+            }
+            getLocationName()
+        }
+
+        private fun getLocationName() {
+            tryToExecute(
+                function = { mobileLocationRepository.getLocationName(state.value.currentLocation.toEntity()) },
```

Comment:
I suggest moving this:
```suggestion
                function = onGenLocationName
```

Reply:
done